File: javamodule/src/week7/section2/AnimalSimulator.java
package week7.section2;

public class AnimalSimulator {
    public static void main(String[] args) {
        Dog aDog = new Dog("진도개", "화랑", 3);
        Dog bDog = new Dog("삼살개", "레오", 2);

        for (int i = 1; i <= 5; ++i) {
            int dogChoice = (int) (Math.random() * 2 + 1);
            Dog dog;
            if (dogChoice == 1)
                dog = aDog;
            else dog = bDog;
            int dogAction = (int) (Math.random() * 4 + 1);
            switch (dogAction) {
                case 1:
                    dog.bark();
                    break;
                case 2:
                    dog.eat();
                    break;
                case 3:
                    dog.sleep();
                    break;
                case 4:
                    dog.move();
                    break;
            }
        }

        Cat aCat = new Cat("페르시안 고양이", "네로", 3);
        Cat bCat = new Cat("샴 고양이", "나비", 5);

        for (int i = 1; i <= 5; ++i) {
            int catChoice = (int) (Math.random() * 2 + 1);
            Cat cat;
            if (catChoice == 1)
                cat = aCat;
            else cat = bCat;
            int catAction = (int) (Math.random() * 4 + 1);
            switch (catAction) {
                case 1:
                    cat.meow();
                    break;
                case 2:
                    cat.eat();
                    break;
                case 3:
                    cat.sleep();
                    break;
                case 4:
                    cat.move();
                    break;
            }
        }
        Duck aDuck = new Duck("집오리", "도널드",2);
        Duck bDuck = new Duck("청둥오리", "하늘",2);

        for (int i = 1; i <= 5; ++i) {
            int duckChoice = (int) (Math.random() * 2 + 1);
            Duck duck;
            if (duckChoice == 1)
                duck = aDuck;
            else duck = bDuck;
            int duckAction = (int) (Math.random() * 5 + 1);
            switch (duckAction) {
                case 1:
                    duck.quack();
                    break;
                case 2:
                    duck.eat();
                    break;
                case 3:
                    duck.sleep();
                    break;
                case 4:
                    duck.move();
                    break;
                case 5:
                    duck.swim();
                    break;
            }
        }
    }
}

class Animal {
    protected String type;
    protected String name;
    protected int age;

    public Animal(String type, String name, int age) {
        this.type = type;
        this.name = name;
        this.age = age;
    }

    public void eat() {
        System.out.println(age + "세의 " + type + "인 " + name + "(이/가) 식사 중입니다");
    }

    public void sleep() {
        System.out.println(age + "세의 " + type + "인 " + name + "(이/가) 잠자는 중입니다");
    }

    public void move() {
        System.out.println(age + "세의 " + type + "인 " + name + "(이/가) 이동 중입니다");
    }
}

class Dog extends Animal {
    public Dog(String type, String name, int age) {
        super(type, name, age);
    }

    public void bark() {
        System.out.println(age+"세의 " +type+ "인 "+name +  "(이/가) 짖고 있습니다");
    }
}

class Cat extends Animal {
    public Cat(String type, String name, int age) {
        super(type, name, age);
    }

    public void meow() {
        System.out.println(age+"세의 " +type+ "인 "+name +  "(이/가) 야옹하고 있습니다");
    }
}

class Duck extends Animal {
    public Duck(String type, String name, int age) {
        super(type, name, age);
    }

    public void quack() {
        System.out.println(age+"세의 " +type+ "인 "+name +  "(이/가) 꽥꽥하고 있습니다");
    }
    public void swim() {
        System.out.println(age+"세의 " +type+ "인 "+name +  "(이/가) 수영하고 있습니다");
    }
}

File: src/main/kotlin/week7/section2/AnimalSimulator.kt
package week7.section2

open class Animal (protected var type: String,
                   protected var name: String,
                   protected val age: Int){

    fun eat() = println("${age}세의 ${type}인 ${name} +(이/가) 식사 중입니다")
    fun sleep() = println("${age}세의 ${type}인 ${name} +(이/가) 잠자는 중입니다")
    fun move() = println("${age}세의 ${type}인 ${name} +(이/가) 이동 중입니다")
}

class Dog(type: String, name: String, age: Int) : Animal(type,name,age) {
    fun bark() = println("${age}세의 ${type}인 ${name} + (이/가) 짖고 있습니다")
}

class Cat(type: String, name: String, age: Int) : Animal(type,name,age) {
    fun meow() = println("${age}세의 ${type}인 ${name} + (이/가) 야옹하고 있습니다")
}

class Duck(type: String, name: String, age: Int) : Animal(type,name,age) {
    fun quack() = println("${age}세의 ${type}인 ${name} + (이/가) 꽥꽥하고 있습니다")
    fun swim() = println("${age}세의 ${type}인 ${name} + (이/가) 수영하고 있습니다")
}

fun main() {
    val aDog = Dog("진도개","화랑",3)
    val bDog = Dog("삼살개","레오",2)

    for (i in 1..5) {
        val dogChoice = (Math.random() * 2 + 1).toInt()
        val dog = if (dogChoice == 1) aDog else bDog
        val dogAction = (Math.random() * 4 + 1).toInt()
        when (dogAction) {
            1 -> dog.bark()
            2 -> dog.eat()
            3 -> dog.sleep()
            4 -> dog.move()
        }
    }

    val aCat = Cat("페르시안 고양이", "네로", 3)
    val bCat = Cat("샴 고양이", "나비", 5)

    for (i in 1..5) {
        val catChoice = (Math.random() * 2 + 1).toInt()
        val cat = if (catChoice == 1) aCat else bCat
        val catAction = (Math.random() * 4 + 1).toInt()
        when (catAction) {
            1 -> cat.meow()
            2 -> cat.eat()
            3 -> cat.sleep()
            4 -> cat.move()
        }
    }

    val aDuck = Duck("집오리", "도널드",2)
    val bDuck = Duck("청둥오리", "하늘",2)

    for (i in 1..5) {
        val duckChoice = (Math.random() * 2 + 1).toInt()
        val duck = if (duckChoice == 1) aDuck else bDuck
        val duckAction = (Math.random() * 5 + 1).toInt()
        when (duckAction) {
            1 -> duck.quack()
            2 -> duck.eat()
            3 -> duck.sleep()
            4 -> duck.move()
            5 -> duck.swim()
        }
    }
}